lib/rucio/core/transfer.py

```diff
@@ -667,6 +667,13 @@ def get_transfer_requests_and_source_replicas(total_workers=0, worker_number=0,
                     else:
                         verify_checksum = 'both'
 
+                logging.info('source RSE checksum compatibility: {}'.format(rse_attrs[source_rse_id][CHECKSUM_KEY]))
+                logging.info('destination RSE checksum compatibility: {}'.format(rse_attrs[dest_rse_id][CHECKSUM_KEY]))
+                common_checksum_names = set(rse_attrs[dest_rse_id][CHECKSUM_KEY]).intersection(rse_attrs[source_rse_id][CHECKSUM_KEY])
+                if len(common_checksum_names) == 0:
+                    logging.info('No common checksum method. Verifying destination only.')
+                    verify_checksum = 'destination'
+
```

Comment:
I think this will break production as it assumes that the `supported_checksums` is set in the `rse_attrs`. But it will not be unless we manually add it everywhere (Which we won't, it should be optional). Thus it will fall back everything to `verify_checksum` destination.

Reply:
That's right, I forgot I have implemented the `get_rse_supported_checksums` method for such usage in `core/rse.py:541`.
I have modified the logic to use such method that returns the custom attribute if set, otherwise the full set of supported checksums.